Jenkins: Add CheriBSD boot testing

arichardson opened the pull request:
This ensures that we can always boot the latest master
branch of CheriBSD.

arichardson committed:
Jenkins: Add CheriBSD boot testing

This ensures that we can always boot the latest master
branch of CheriBSD.

arichardson commented:
Hmm debug+asan times out while booting. Should we skip it or increase the timeouts?

arichardson commented:
Should hopefully pass now after https://github.com/CTSRD-CHERI/qemu/commit/654af07968265d4a3c3f1ec6ac963a7feff21e78

jrtc27 commented on Jenkinsfile:
```diff
@@ -56,6 +56,48 @@ def maybeArchiveArtifacts(params, String os) {
+def bootCheriBSDForAllArchitectures(params, String config) {
+    stage("Boot CheriBSD (${config})") {
+        bootJobs = [failFast: false]
+        // TODO: should we also boot other images?
```
Do you mean the MFS_ROOT ones, or hybrid/vanilla images? Not sure hybrid really adds much, it should be tested enough by hybrid kernels. Vanilla is probably more interesting to check we didn't break things like setting xEPC rather than xEPCC.

arichardson replied:
That sounds good. We might also want to run the qemu tests that boot Linux etc.

jrtc27 commented on Jenkinsfile:
```diff
@@ -56,6 +56,48 @@ def maybeArchiveArtifacts(params, String os) {
+        bootJobs = [failFast: false]
+        // TODO: should we also boot other images?
+        ["riscv64-purecap", "morello-purecap", "mips64-purecap"].each { String architecture ->
+            // TODO: boot both purecap and hybrid rather than just the default kernel
```
Yeah this would be good to have

jrtc27 commented on Jenkinsfile:
```diff
@@ -83,8 +125,10 @@ selectedConfigs.each { config ->
                         extraBuildSteps(params, os)
-                        // Don't archive the Debug+ASAN artifacts
+                        // For Debug+ASAN we don't archive artifacts.
+                        // We also skip the boot test (for now) since the test script times out.
```
Any idea how long it would take to run the tests with a Debug+ASAN QEMU? It'd be nice to have, but not if it takes an hour...

arichardson replied:
Haven't tried yet, hopefully only 3-5 times slower. Might even be similar since the LTO build takes so much longer.

chriszyx committed:
Fix aarch64 debug register names.

Give all the debug registers their correct names including the
index, rather than having multiple registers all with the
same name string, which is confusing when viewed over the
gdbstub interface.

Signed-off-by: CHRIS HOWARD <cvz185@web.de>
Reviewed-by: Richard Henderson <richard.henderson@linaro.org>
Message-id: 4127D8CA-D54A-47C7-A039-0DB7361E30C0@web.de
[PMM: expanded commit message]
Signed-off-by: Peter Maydell <peter.maydell@linaro.org>
(cherry picked from commit e1be11a5a440bfa552e0a8bb109d72294054e3f0)

arichardson committed:
Ensure we clean up artifacts after running tests

arichardson committed:
Remove a warning message that causes tests to fail

arichardson committed:
Revert "Add better games for AARCH64 DBG registers"

This introduces a memory leak and was fixed correctly upstream.

This reverts commit 31b7939bb7dbb0782c55060858c343ebcf69721f.
